internal/manifests/manifests.go

```diff
@@ -167,18 +163,30 @@ func (m *Manifests) ListClusterManifestsInternal(ctx context.Context, params ope
 		return nil, common.NewApiError(http.StatusInternalServerError, err)
 	}
 
+	// legacyUserManifestPaths is temporarily included until no more legacy manifest paths are supported
+	legacyUserManifestPaths, err := m.FindUserManifestPathsByLegacyMetadata(ctx, strfmt.UUID(params.ClusterID.String()))
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to identify user manifest paths based on filesystem")
+	}
+
 	manifests := models.ListManifests{}
 	for _, file := range files {
-		folder, filename, err := ParsePath(file)
+		folder, filename, err := ParsePath(file.Path)
 		if err != nil {
 			return nil, err
 		}
-		isUserManifest, err := m.IsUserManifest(ctx, params.ClusterID, folder, filename)
-		if err != nil {
-			return nil, err
+
+		manifestSource := constants.ManifestSourceSystemGenerated
+		manifestSourceAttributeValue, ok := file.Metadata[constants.ManifestSourceAttribute]
+		if ok && manifestSourceAttributeValue != nil {
+			manifestSource = *manifestSourceAttributeValue
 		}
-		if isUserManifest || swag.BoolValue(params.IncludeSystemGenerated) {
-			manifests = append(manifests, &models.Manifest{FileName: filename, Folder: folder})
+		// If we found no manifest source information in metdata, check the filesystem (legacy method) for an override.
+		if manifestSourceAttributeValue == nil && swag.ContainsStrings(legacyUserManifestPaths, file.Path) {
+			manifestSource = constants.ManifestSourceUserSupplied
+		}
+		if manifestSource == constants.ManifestSourceUserSupplied || swag.BoolValue(params.IncludeSystemGenerated) {
+			manifests = append(manifests, &models.Manifest{FileName: filename, Folder: folder, ManifestSource: &manifestSource})
```

Comment:
In rare case, it might be a breaking change for some client (that doesn't allow unknown properties)

I don't know how we handle thoses cases so far

Reply:
We will not have any breaking change because of this, the instruction is clear

```
if manifestSource == constants.ManifestSourceUserSupplied || swag.BoolValue(params.IncludeSystemGenerated) {
			manifests = append(manifests, &models.Manifest{FileName: filename, Folder: folder, ManifestSource: &manifestSource})
		}
```

If the manifest is deemed to be user supplied (if not, we do not care) then it is to be listed.

Reply:
I have certainly already tested this code in all of the use cases we have for it

API and UI,

There is no breakage within assisted because of this change.

This change also simplifies the handling of log filenames (better separation of concerns) and opens up the possibility of making the UI more user friendly, for these reasons I would like to add this extra field.

We can ask @avishayt if he would be OK with this change. I personally don't think the addition of a field should cause a problem, but let's make sure it's not an issue.

Reply:
I spoke with @avishayt and he is OK with us adding the field.